### ISSUE TITLE
Model checks against simulation

### ISSUE THREAD

**Issue by seabbs** [epiforecasts/bp.delta#7]
Check the model works against some basic simulations generated in R. 

Implement by: 

- Adding a `generative_model()` function or similar for generating data
- Adding some scenarios to evaluate potentially as a package loaded data source (i.e and therefore fixed)
- Add model fitting to these scenarios in the `inst/forecasts` folder
- Add evaluation of these fits in the `inst/evaluation` folder (likely just posterior predictive plots and parameter posteriors compared to input parameters)

**Comment by jbracher** [comment]
For a paper one idea could be to simulate from models with the same parameters as obtained in the real-data study (assuming we cover different countries). Not entirely sure how to do that, though (I assume fixing parameters at their posterior means?).

**Comment by seabbs** [comment]
Yes good idea. I am looking at some simplistic simulation based calibration as a first step in.
